contents/testing/integration-testing.md

```diff
@@ -0,0 +1,212 @@
+Using the example shown above, below shows a sample integration test used for the `login` and `ordering` modules:
+
+~~~
+Test case ID: 1
+Objective: Check the link between login and ordering modules
+Description: Enter login credentials and click on login button
+Expected result: To be directed to order food page based on the login user
+~~~
+
+<box type="warning">As integration test cases are <tooltip content="It generally takes longer time due to additional overheads such as waiting for dependent modules to respond">expensive operations</tooltip> compared to unit testing, it should focus mainly on the integration of modules together and not on specific actions within the same module.</box>
+
+:fas-info-circle: More details about the ways to structure incremental testing can be found on [this article](https://www.softwaretestinggenius.com/various-approaches-in-integration-testing/)
+
+### 3. Executing test scripts and reporting defects
+
+Depending on the approach you have chosen for your integration plan and the test cases, the way you execute your code for testing will differ.
+
+- Big bang approach usually involves all the modules to be developed before you can start with the integration testing.
+
+- Incremental approach is usually conducted simultaneously with the module's development. Stubs and drivers are used to mimic the modules for writing tests since not all modules are developed at the point of integration testing.
+[NUS Software Engineer textbook](https://nus-cs2103-ay1819s1.github.io/cs2103-website/se-book-adapted/chapters/testing.html#stubs) provides concrete codes for using stubs.
+
+Either way, you should always ensure that all high prioritized bugs are fixed before moving on.
+
+:fas-info-circle: [This article](https://www.guru99.com/test-environment-software-testing.html) shows more details on how to set up a test environment for better integration testing.
+
+### 4. Tracking defects and re-testing application
+
+In the event of failing your integration test case, it is important to learn how to track down the [<tooltip content="Incorrect behavior observed from the system">defects</tooltip>](https://qacomplete.com/resources/articles/what-is-a-software-defect/) occured. 
+Thereafter, you should make changes to your application to fix them and re-test your application with integration testing to ensure that the defects are no longer there.
+
+:fas-info-circle: [This article](http://www.professionalqa.com/defect-tracking-process) covers more details on how to effectively track down defects in a system and fix them.
+
+## Tips for better integration tests
+
+### :fas-thumbs-up: Make sure that each component is unit tested before integration testing
+By ensuring that each unit test is completed properly, integration testing will be smoother as we can focus mainly on the flow of data between modules.
+
```

Comment:
As these sections are quite small, format as bullet points instead? e.g.,

* **Make sure that each component is unit tested before integration testing**
  By ensuring that each unit test is completed properly, integration testing will be smoother as we can focus mainly on the flow of data between modules.